### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by cmplx-xyttmt**
Created API endpoints for rides, ride requests, create rides and create ride requests.
Feedback is appreciated.

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[run]
+source=
+    ./
+omit=
+    ./venv/*
```
Comment (fnshemerirwe) [inline .coveragerc]: Please add a new line at the end of this file

File: .coveralls.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+repo_token: PgqSvXaAwAZ7HB8YknZkNVNmjFMQcTALW
```
Comment (fnshemerirwe) [inline .coveralls.yml]: Please add a new line at the end of this file

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.idea
+venv/
+app/__pycache__/
```
Comment (fnshemerirwe) [inline .gitignore]: Please add a new line at the end of this file

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+language: python
+python: 3.5.2
+script: python -m unittest discover && coverage run -m unittest discover && coverage report
+after_success: coveralls
```
Comment (fnshemerirwe) [inline .travis.yml]: Please add a new line at the end of this file

File: Procfile [hunk 5]
```diff
@@ -0,0 +1 @@
+web gunicorn run:app
```
Comment (fnshemerirwe) [inline Procfile]: Please add a new line at the end of this file

File: app/app.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+@app.route('/ridemyway/api/v1/rides', methods=['GET'])
+def get_rides():
+    rides_as_dicts = [convert_ride_offer(ride) for ride in rides]
+    return jsonify({'rides': rides_as_dicts})
+
+
+@app.route('/ridemyway/api/v1/rides/<int:ride_id>', methods=['GET'])
+def get_ride(ride_id):
+    if ride_id > len(rides):
+        abort(404)
```
Comment (fnshemerirwe) [inline app/app.py]: Correct validations but could you please add specific error messages for each condition.

File: app/app.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    """Converts ride offer to json serializable object by first converting requests to dict object"""
+    ride_requests_list = [ride_req.__dict__ for ride_req in ride_offer.requests]
+    ride_offer_dict = ride_offer.__dict__
+    ride_offer_dict["requests"] = ride_requests_list
+    return ride_offer_dict
+
+
+@app.route('/ridemyway/api/v1/rides', methods=['POST'])
+def create_ride():
+    if not request.is_json:
```
Comment (fnshemerirwe) [inline app/app.py:36]: Correct validations but could you please add specific error messages for each condition.

File: app/app.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+def get_rides():
+    rides_as_dicts = [convert_ride_offer(ride) for ride in rides]
+    return jsonify({'rides': rides_as_dicts})
+
+
+@app.route('/ridemyway/api/v1/rides/<int:ride_id>', methods=['GET'])
+def get_ride(ride_id):
+    if ride_id > len(rides):
+        abort(404)
+    ride = rides[ride_id - 1]
```
Comment (fnshemerirwe) [inline app/app.py]: What happens if my ride_id <= 0?

**Review by fnshemerirwe — CHANGES_REQUESTED** [review]
Please add http status codes to your responses

File: app/app.py (hunk 8 above)
Comment (cmplx-xyttmt) [reply app/app.py]: Oh. I'll add a condition to handle that case.

Comment (cmplx-xyttmt) [reply app/app.py]: I've handled the edge case. Thanks for pointing it out.

File: app/app.py (hunk 7 above)
Comment (cmplx-xyttmt) [reply app/app.py:36]: Added specific error messages

File: app/app.py (hunk 6 above)
Comment (cmplx-xyttmt) [reply app/app.py]: Added specific error messages

File: Procfile (hunk 5 above)
Comment (cmplx-xyttmt) [reply Procfile]: Added the  new line

File: .travis.yml (hunk 4 above)
Comment (cmplx-xyttmt) [reply .travis.yml]: Added the  new line

File: .coveragerc (hunk 1 above)
Comment (cmplx-xyttmt) [reply .coveragerc]: Added the  new line

File: .coveralls.yml (hunk 2 above)
Comment (cmplx-xyttmt) [reply .coveralls.yml]: Added the  new line

File: .gitignore (hunk 3 above)
Comment (cmplx-xyttmt) [reply .gitignore]: Added the  new line

**Review by fnshemerirwe — CHANGES_REQUESTED** [review]
Using `abort(400)` to handle edge cases is good, but as an end user, I can't tell what caused the bad request for example,
```if ride_id > len(rides) or ride_id <= 0:
       abort(404)```
you can return an informative error message in your responses.

**Review by fnshemerirwe — APPROVED** [review]

**Pull request merged**